pkg/virt-controller/services/template.go

```diff
@@ -612,6 +612,33 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 		}
 	}
 
+	for _, accessCred := range vmi.Spec.AccessCredentials {
+		secretName := ""
+		if accessCred.SSHPublicKey != nil && accessCred.SSHPublicKey.Source.Secret != nil {
+			secretName = accessCred.SSHPublicKey.Source.Secret.SecretName
+		} else if accessCred.UserPassword != nil && accessCred.UserPassword.Source.Secret != nil {
+			secretName = accessCred.UserPassword.Source.Secret.SecretName
+		}
+
+		if secretName == "" {
+			continue
+		}
+		volumeName := secretName + "-access-cred"
```

Comment:
I see that we use this pattern all over the place. Probably not the best best solution. Seems like we can end up with collisions. Since we never generate volumes for users if they are not in the `volumes` list of the VMI, I wonder why we ever started with adding suffixes. :/

Reply:
In this case, the access credential secret name is not a volume, so the suffix is actually meant to help reduce collisions. Using the secret name + suffix allows us to easily infer the secret's direction from within the pod. 

any thoughts on what might be a potentially better solution?